app/store/actions/healthcareAuthorities/getHealthcareAuthoritiesAction.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
```

Comment:
started the review before you merged some updates yesterday, so this maybe isnt relevant to this pr anymore, but still interested in your thoughts here!

Reply:
IMO, the major con is that is requires having a dedicated type files to facilitate the export (which I set up here to allow for the explicit export type). 
It's kinda a pain on HOF exports like the thunks, but now that its setup it should be straightforward going forward.

Reply:
> IMO, the major con is that is requires having a dedicated type files to facilitate the export.

Im not sure this is true? my understanding of the rule is that the any export in any ts file needs to be explicitly typed. The file doesnt need to have only types declared in it.

ref:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/explicit-module-boundary-types.md

Reply:
sorry, unclear explanation on my part -- 
I mean to avoid circular dependencies / keeping the types in a sensible place.

E.g., for something like `type HealthcareAuthority`,there are three locations that it would make sense to have the type definition: 
1. the API file which receives the payload from the API, and returns the `HealthcareAuthority` type
2. the action layer, which deals with ensuring correct types are sent to the store
3. the reducer, which deals with ensuring the store has a consistent type. (Declare the type alongside store shape)

1. Is how I typically structure imports without types: API -> Action -> Reducer inheritance, because then the API is a leaf. However, in a type system then components who have methods on the type are importing from the API (which imo is ideally abstracted away if they interface via the actions). E.g. 
```js
// This component file is aware of both the API layer and the Redux layer
import { FooSelector } from "selectors";
import { FooType } from "API";

const localFooTransform = (foo: FooType): FooType => { ... };

const ExampleComponent = () => {
  const foo = select(FooSelector);
};
```
That said, my solution of the API taking the types from `../store/types` isn't much better, as it still becomes aware of redux. 

2. & 3. bring circular dependencies (which aren't a huge deal, but imo reflects a confusing abstraction)

Without a root types file, where would you recommend defining types? I don't work with typescript much so I'm curious if there's a better pattern for share types between abstraction layers

Reply:
I think the first design consideration is whether or not the API is what is dictating the shape of the types.  If you dont control the api, or its a bit hairy, i usually see a distinction between the type that the api returns and the type that is used throughout the application.  and a function to convert between the two.

so you might have something like: 

```
/api
  /networkModel
     healthcareAuthority.ts
     foobar.ts
     index.ts
  /networkConverters
     healthcareAuthority.ts
     fooboar.ts
     index.ts
   /dataModel
     healthcareAuthority.ts
     foobar
     index.ts
api.ts
```

and you can use some module importing to access them pretty conveniently

```typescript
//api.ts
import { HealthcareAuthority } from './networkModel'

const fetchHealthAuthorities: = async (): Promise<NetworkResponse<HealthcareAuthority[]>> => {
   const reponse = fetch(url, ...
    const networkData = handle the response
   return networkData
...
```

```typscript
// reducer.ts or action.ts

import { HealthAuthority } from "../api/dataModel"
import { toHealthAuthority } from "../api/networkConverter"

const getHealthcareAuthorities_success = createAction(
  GET_HEALTHCARE_AUTHORITIES_SUCCESS,
  ({ data }) => ({ payload: { healthcareAuthorities: data.map(toHealthcareAuthority) } }),
);

```

so you can still pull out the types to a dedicated file, but its just grouped by feature/domain concept, rather than the fact it is a type.


And you if you are basing the type off of the api, you could simply have the dataModel folder or just have the type files at the same level as everything else.

Reply:
Good points on the data model differences between the layers! that makes a lot of sense, thanks